### ISSUE TITLE
icon: multiple SVGs on a page should have unique IDs

### ISSUE THREAD

**Issue by coennijhuis** [angular/material#11395]
### BUG:
When adding multiple identical SVGs to a page, they should all have unique IDs.

### CodePen and steps to reproduce the issue:

CodePen link: https://codepen.io/anon/pen/jpxBZx

### Detailed Reproduction Steps: 
1. Click the CodePen link
2. Check the icons in the preview pane. There are 5 identical icons. 
3. Open developer toolbar (in your browser of choice)
4. Inspect the icons. 
5. Check the ID inside the g tag are all the same. (these should all be unique)

### What is the expected behavior?
The expected behavior is that all the IDs in the <g> tags are unique.

### What is the current behavior?
The current behavior is that the IDs are all the same

### What is the use-case or motivation for changing an existing behavior?
Accessibility standards state that an ID should only be used once in page. (WCAG) 

### Which versions of AngularJS, Material, OS, and browsers are affected?

- AngularJS: 1.7.2
- AngularJS Material: 1.1.10
- OS: all
- Browsers: all
 
### Is there anything else we should know? Stack Traces, Screenshots, etc.
NA

**Comment by Splaktar** [comment]
Duplicate of https://github.com/angular/material/issues/8689, but that one did not include a CodePen demo, so thank you very much for that. I'll copy the link to the CodePen over to that issue. 

It looks like the proposed PR fix in https://github.com/angular/material/pull/11342 does not solve the problem in this CodePen. I'll investigate some more.

**Comment by Splaktar** [comment]
Re-opening this as the id mangling in #11342 is not the same as the lack of id caching in this issue.

**Comment by Splaktar** [comment]
This has been a problem since id caching was first implemented in `1.0.7`.

https://github.com/angular/material/pull/11465#issuecomment-429327498 found that all of the icons loaded on the initial page load are loaded without caching. Caching then starts after the page load has been completed.
